### DISCUSSION TITLE
Updating/connecting a parent in a children/parent relation

### DISCUSSION THREAD

**Discussion by jacobclarke92** [prisma/prisma#8681] [Q&A] [answered]
I have schema that looks like this:
```prisma
model Page {
  id           Int        @id @default(autoincrement())
  title        String?
  parentId     Int?
  parent       Page?      @relation("Parent", fields: [parentId], references: [id])
  children     Page[]     @relation("Parent")
}
```

When updating a page I want to be able to connect the parent and/or update parent data
e.g.
```typescript
prisma.page.update({
  where: { id: 1 },
  data: {
    parent: {
      connect: { id: 2 },
      update: { title: 'Renamed Page' },
    },
  },
})
```

But I get an error that seems to imply the only option is to update `parentId` and remove `parent` from the post data altogether.
```
Unknown arg `parent` in data.parent for type PageUncheckedUpdateInput. Did you mean `parentId`?
```

Am I missing something obvious or doing something dumb?
The weird thing is I get an autocomplete suggestion that implies it should be possible, but this could just be a weird inconsistency to do with nested types or something:
<img width="849" alt="Screen Shot 2021-08-11 at 2 15 12 pm" src="https://user-images.githubusercontent.com/9414552/128968842-35741b7e-abc6-4e53-8678-68122ac18333.png">

EDIT: I forgot to mention I'm using prisma 2.28.0

**Comment by ryands17** [comment, marked as answer]
@jacobclarke92 👋 

I tried this on a clean database with the above schema and it works for me. Could you check again with the latest version of Prisma or provide a sample reproduction if possible?

**Reply by jacobclarke92** [reply]
Apologies I think I might have wasted your time. Thanks for the speedy response though!

My issue was unrelated and led me to believe the issue was with Prisma when it wasn't -- it was due to an issue with [next-crud](https://github.com/premieroctet/next-crud)'s implementation of their prisma adaptor.

It honestly looks like it should all work seeing as it just passes request post data directly into prisma update data value
https://github.com/premieroctet/next-crud/blob/2cfa1d90d3c5fa8673d09042108fde648973d75b/src/handler.ts#L139
https://github.com/premieroctet/next-crud/blob/2cfa1d90d3c5fa8673d09042108fde648973d75b/src/handlers/updateHandler.ts#L22

The reason it fails must just be due to faulty generic typings or something.
This is the raw output for reference:
```

Invalid `adapter.update()` invocation in
/Users/jacobclarke/Projects/gita/cms-next/node_modules/@premieroctet/next-crud/dist/handlers/updateHandler.js:21:51

  18 return __awaiter(this, void 0, void 0, function* () {
  19     const resource = yield adapter.getOne(resourceId, query);
  20     if (resource) {
→ 21         const updatedResource = yield adapter.update(resourceId, body, query){
               where: {
                 id: 1
               },
               data: {
                 parent: {
                 ~~~~~~
                   connect: {
                     id: 2
                   }
                 },
                 title: 'Manually updated',
                 slug: 'home',
                 heading: null,
                 lead: null,
                 description: null,
                 metaTitle: null,
                 metaDescription: null,
                 shareTitle: null,
                 shareDescription: null,
                 showInMenu: true,
                 showInHeader: false,
                 showInFooter: false,
                 published: true,
                 publishAt: null,
                 expireAt: null,
                 categories: {
                   connect: [
                     {
                       id: 1
                     },
                     {
                       id: 6
                     },
                     {
                       id: 8
                     }
                   ]
                 },
                 id: 1,
                 uuid: 'ckrpyyoq70000k1fc3x1h9zf1',
                 order: 0,
                 lft: 0,
                 rght: 0,
                 parentId: 2,
                 imageId: null,
                 path: '/home',
                 createdAt: '2021-07-30T06:33:04.686Z',
                 updatedAt: '2021-08-13T00:15:12.441Z'
               },
               select: undefined,
               include: {
                 parent: true,
                 image: true,
                 categories: true
               }
             })

Unknown arg `parent` in data.parent for type PageUncheckedUpdateInput. Did you mean `parentId`? Available args:
type PageUncheckedUpdateInput {
  id?: Int | IntFieldUpdateOperationsInput
  uuid?: String | StringFieldUpdateOperationsInput
  order?: Int | NullableIntFieldUpdateOperationsInput | Null
  lft?: Int | NullableIntFieldUpdateOperationsInput | Null
  rght?: Int | NullableIntFieldUpdateOperationsInput | Null
  parentId?: Int | NullableIntFieldUpdateOperationsInput | Null
  imageId?: Int | NullableIntFieldUpdateOperationsInput | Null
  path?: String | NullableStringFieldUpdateOperationsInput | Null
  title?: String | StringFieldUpdateOperationsInput
  slug?: String | NullableStringFieldUpdateOperationsInput | Null
  heading?: String | NullableStringFieldUpdateOperationsInput | Null
  lead?: String | NullableStringFieldUpdateOperationsInput | Null
  description?: String | NullableStringFieldUpdateOperationsInput | Null
  metaTitle?: String | NullableStringFieldUpdateOperationsInput | Null
  metaDescription?: String | NullableStringFieldUpdateOperationsInput | Null
  shareTitle?: String | NullableStringFieldUpdateOperationsInput | Null
  shareDescription?: String | NullableStringFieldUpdateOperationsInput | Null
  showInMenu?: Boolean | BoolFieldUpdateOperationsInput
  showInHeader?: Boolean | BoolFieldUpdateOperationsInput
  showInFooter?: Boolean | BoolFieldUpdateOperationsInput
  published?: Boolean | BoolFieldUpdateOperationsInput
  publishAt?: DateTime | NullableDateTimeFieldUpdateOperationsInput | Null
  expireAt?: DateTime | NullableDateTimeFieldUpdateOperationsInput | Null
  createdAt?: DateTime | DateTimeFieldUpdateOperationsInput
  updatedAt?: DateTime | DateTimeFieldUpdateOperationsInput
  children?: PageUncheckedUpdateManyWithoutParentInput
  categories?: PageCategoriesUncheckedUpdateManyWithoutPageInput
}
```

Aaanyway, all good. I'll probably just ditch next-crud altogether and write a less generic solution for my needs.
But if you notice anything obvious about the above let me know.

**Reply by ryands17** [reply]
This is because you're passing the `parentId` as well. You already have `parent: { connect: { id: 2 } }` so you don't need the explicit `parentId: 2`.

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
